File: gdx2d-desktop/src/hevs/gdx2d/demos/simple/mistify/DemoLines.java
package hevs.gdx2d.demos.simple.mistify;

import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * A classical Mistify screen saver clone.<br>
 * <p/>
 * Code adapted from <a href="http://r3dux.org/2010/11/mystify-2-0/"> http://r3dux.org/2010/11/mystify-2-0</a>
 *
 * @author Pierre-André Mudry
 * @version 1.0
 * @see <a href="http://r3dux.org/2010/11/mystify-2-0/"> http://r3dux.org/2010/11/mystify-2-0</a> for the original source code
 */
public class DemoLines extends PortableApplication {
	final int N_SHAPES = 3;
	BounceShape[] s;
	int frame = 0;

	public static void main(String[] args) {
		new DemoLines();
	}

	@Override
	public void onInit() {
		// Sets the window title
		setTitle("Mistfy screensaver clone, mui 2013");

		// Allocate size for the shapes
		s = new BounceShape[N_SHAPES];

		// Initialize them
		for (int i = 0; i < N_SHAPES; i++) {
			s[i] = new BounceShape(this.getWindowWidth(), this.getWindowHeight());
		}
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {

		// Clears the screen
		g.clear();

		// Draws and moves the shapes
		for (BounceShape shape : s) {
			shape.drawShape(g);
			shape.moveShape(g.getScreenWidth(), g.getScreenHeight());
			shape.shiftShapeColour(frame++);
		}

		/**
		 *  TODO it would be nice to have something like an accumulation buffer that fades here
		 */
		g.drawFPS();
		g.drawSchoolLogo();
	}

}

File: gdx2d-demoDesktop/src_to_do/main/kotlin/ch/hevs/gdx2d/demos/simple/mistify/DemoLines.kt
package ch.hevs.gdx2d.demos.simple.mistify

import ch.hevs.gdx2d.desktop.PortableApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * A classical Mistify screen saver clone.<br></br>
 *
 *
 * Code adapted from [ http://r3dux.org/2010/11/mystify-2-0](http://r3dux.org/2010/11/mystify-2-0/)
 *
 * @author Pierre-André Mudry
 * @version 1.0
 * @see [ http://r3dux.org/2010/11/mystify-2-0](http://r3dux.org/2010/11/mystify-2-0/) for the original source code
 */
class DemoLines : PortableApplication() {
    val N_SHAPES = 3
    lateinit var s: Array<BounceShape>
    var frame = 0

    override fun onInit() {
        // Sets the window title
        setTitle("Mistfy screensaver clone, mui 2013")

        // Allocate size for the shapes
        s = Array(N_SHAPES){BounceShape(this.windowWidth, this.windowHeight)}
    }

    override fun onGraphicRender(g: GdxGraphics) {

        // Clears the screen
        g.clear()

        // Draws and moves the shapes
        for (shape in s) {
            shape.drawShape(g)
            shape.moveShape(g.screenWidth, g.screenHeight)
            shape.shiftShapeColour(frame++)
        }

        /**
         * TODO it would be nice to have something like an accumulation buffer that fades here
         */
        g.drawFPS()
        g.drawSchoolLogo()
    }
}

fun main(args: Array<String>) {
  DemoLines()
}
